File: Year-2023/LeetCode-August-Challenge/UniquePaths.java
import java.util.*;

class Solution {
    public int uniquePaths(int m, int n) {
        int[][] dp = new int[m][n];

        for (int[] x : dp)
            Arrays.fill(x, 1);

        for (int i = 1; i < m; ++i)
            for (int j = 1; j < n; ++j)
                dp[i][j] = dp[i - 1][j] + dp[i][j - 1];

        return dp[m - 1][n - 1];
    }
}

public class UniquePaths {
    public static void main(String[] args) {
        int m = 3, n = 7;
        System.out.println(new Solution().uniquePaths(m, n));
    }
}

File: Year-2023/LeetCode-August-Challenge/UniquePaths.js
var uniquePaths = function (m, n) {
    let dp = new Array(m);
    for (let i = 0; i < dp.length; ++i)
        dp[i] = new Array(n).fill(1);

    for (let i = 1; i < m; ++i)
        for (let j = 1; j < n; ++j)
            dp[i][j] = dp[i - 1][j] + dp[i][j - 1];

    return dp[m - 1][n - 1];
};

let m = 3, n = 7;
console.log(uniquePaths(m, n));